config/configuration_singleton.rb

```diff
@@ -36,6 +36,16 @@ def load_external_config?
     to_bool(ENV['OOD_LOAD_EXTERNAL_CONFIG'] || (rails_env == 'production'))
   end
 
+  # The batch_connect apps' configuration root directory
+  def bc_config_root
+    Pathname.new(ENV["OOD_BC_APP_CONFIG_ROOT"] || "/etc/ood/config/apps")
+  end
+
+  # Whether we load the batch_connect apps' configuration from external root
+  def load_external_bc_config?
+    to_bool(ENV["OOD_LOAD_EXTERNAL_BC_CONFIG"])
+  end
```

Comment:
Before I add comments, can you further explain your thinking on the intention of `load_external_bc_config?` and `bc_config_root` and then the logic when actually loading?

For example, the implementation of

    def load_external_config?
       to_bool(ENV['OOD_LOAD_EXTERNAL_CONFIG'] || (rails_env == 'production'))
     end

has these goals:

1. default functionality is `load_external_config?` is true when `rails_env == 'production'`
2. if the OOD_LOAD_EXTERNAL_CONFIG env is set we use the bool value of that string, whether it is yes/no 1/0 true/false. This allows me in Rails development mode to enable loading external config, and in Rails production mode disable loading external config using a .env.local file. The former more than the latter is really useful for development.

It seems that `load_external_bc_config?` doesn't yet serve that purpose because it is always false unless `OOD_LOAD_EXTERNAL_BC_CONFIG` is set to something; and if `load_external_bc_config?` is false we are still loading external config for sys apps if the sys app directory exists:

    if Configuration.load_external_bc_config? || (router.type == :sys && global_sub_app_root.directory?)

Reply:
Ah, I forgot that when the env var is `nil` we fallback to default decision-making, I am calling the `to_bool` too early here. I will need to rethink this.

The simplest solution is that I remove the `#load_external_bc_config?` method and just use:

```ruby
if router.type == :sys && global_sub_app_root.directory?
```